### PR TITLE
Gsoc 2020

### LINKED ISSUE #705: [NG] Migrate the site data for the GSoC page to the layer5-ng site
**Description**
Currently, the internship page dedicated to [GSoC](https://layer5ng.netlify.app/careers/programs/gsoc/2020) on the [layer5-ng site](https://layer5ng.netlify.app/) points to a 404 error. 

**Expected Behavior**

The data from the [GSoC page](https://layer5.io/programs//gsoc/2020) on the [layer5.io site](https://layer5.io/) needs to be migrated to the layer5-ng site. 

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by vinayaksh42**
Signed-off-by: Vinayak Sharma <vinayaksh42@gmail.com>

**Description**
Adds the Gsoc 2020 page to layer5-ng
screenshots:
screencapture-localhost-8000-careers-2020-09-15-22_24_40

To migrate Gsoc 2020 page from previous website to layer5ng

previous gsoc 2020 page:
screencapture-layer5-io-programs-gsoc-2020-2020-09-15-22_24_46

This PR fixes #705 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [*] Yes, I signed my commits.

### CODE REVIEW

**Comment by Nikhil-Ladha** [comment]
What is the URL to check it out?

**Comment by vinayaksh42** [comment]
@Nikhil-Ladha i actually made a seprate js file in pages section to develop the content but it isn't linked with anything because i was told to do this: [https://layer5io.slack.com/archives/C015QJKUMPU/p1600172877121000?thread_ts=1600171481.120900&cid=C015QJKUMPU](url)

**Comment by Nikhil-Ladha** [comment]
> @Nikhil-Ladha i actually made a seprate js file in pages section to develop the content but it isn't linked with anything because i was told to do this: https://layer5io.slack.com/archives/C015QJKUMPU/p1600172877121000?thread_ts=1600171481.120900&cid=C015QJKUMPU

Okay, but you yourself must have checked this page somehow, right?

**Comment by vinayaksh42** [comment]
> > @Nikhil-Ladha i actually made a seprate js file in pages section to develop the content but it isn't linked with anything because i was told to do this: https://layer5io.slack.com/archives/C015QJKUMPU/p1600172877121000?thread_ts=1600171481.120900&cid=C015QJKUMPU
> 
> Okay, but you yourself must have checked this page somehow, right? From where did you take this ss?

I made a temporary careers.js file in the pages folder but i removed it from my commit as it was not required.

**Comment by Nikhil-Ladha** [comment]
It would be good if for the time being, you can link this page with the `GSoC20` link, available in the footer.

**Comment by vinayaksh42** [comment]
> It would be good if for the time being, you can link this page with the `GSoC20` link, available in the footer.
> 
> Later on the link will be changed, but for a reference it would be good for now.

okay i will add back the temp page i made and link it with the footer

**Comment by vinayaksh42** [comment]
> It would be good if for the time being, you can link this page with the `GSoC20` link, available in the footer.
> 
> Later on the link will be changed, but for a reference it would be good for now.

okay will add a temp page just for testing.

**Comment by vinayaksh42** [comment]
added a link for testing. @Nikhil-Ladha access it using the footer or by "/gsoc"

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
Thank you gor this @vinayaksh42 . Few changes needs to be addressed:
-  The page is spilling over on the right side in mobile view. Attaching ss.
Screenshot_20200916-013808__01
Screenshot_20200916-013759__01
- The header of the page can be changed to 'Google Summer of Code 2020 and Layer5`
- The above line can be removed from the starting of the description section.
- The line `2020 Program timeline` can be changed to `GSoC 2020 Program Timeline`

**Comment by vinayaksh42** [comment]
@Nikhil-Ladha 
Text changes to the header and "2020 Program timeline".
spilling over to right side in mobile view also fixed
screenshot:
screencapture-localhost-8000-gsoc-2020-09-16-02_18_18

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]
Better to add some padding in between the lines for the heading 
Screenshot_20200916-125111__01
In the below image, the font size of `Project Ideas`  should be same as that of `SMI Conformance Testing`, also the similar subheadings in blue color.
The `Link` text should be avoided and better if the complete link can be used. What I understand here is that on showing the complete link the spilling is occurring so, take care of that accordingly.
Screenshot_20200916-125114__01

File: src/sections/Gsoc/index.js [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+      and strength of our community.
+      </p>  
+
+  <div
+    id="gsoc-2020"
+    className="card-content"
+    style={{position: 'relative', float: 'center', verticalAlign: 'bottom'}}
+  >
+    <h3 style={{textAlign: 'center', color: 'aliceblue'}}>
+      <a
```
Comment (Tanuj22) [inline src/sections/Gsoc/index.js:189]: Use the button defined in `reusecore`.

**Review by Tanuj22 — CHANGES_REQUESTED** [review]
Good work @vinayaksh42 . Can you avoid using inline styles and define classes in the `gsoc.style.js` file itself?

File: src/sections/Gsoc/index.js (hunk 1 above)
Comment (vinayaksh42) [reply src/sections/Gsoc/index.js:189]: okay will implement it.

**Comment by vinayaksh42** [comment]
> The `Link` text should be avoided and better if the complete link can be used. What I understand here is that on showing the complete link the spilling is occurring so, take care of that accordingly.

We can use the <wbr> tag in this case.

**Comment by Nikhil-Ladha** [comment]
> > The `Link` text should be avoided and better if the complete link can be used. What I understand here is that on showing the complete link the spilling is occurring so, take care of that accordingly.
> 
> We can use the "wbr" tag in this case.

Do confirm, if it works well with links too.

**Comment by vinayaksh42** [comment]
> > > The `Link` text should be avoided and better if the complete link can be used. What I understand here is that on showing the complete link the spilling is occurring so, take care of that accordingly.
> > 
> > 
> > We can use the "wbr" tag in this case.
> 
> Do confirm, if it works well with links too.

yes it is working perfefctly, 
kindly check the subheading changes. is it according to what you requested ?

**Comment by vinayaksh42** [comment]
@Tanuj22 Removed inline styles and also used the reusecore button

**Comment by Nikhil-Ladha** [comment]
Now, it's better. Just a nit pick, keep the gray color for the subheadings which had it earlier. I just wanted you to increase the font size for the blue headings and keep the gray ones as it is.

**Comment by vinayaksh42** [comment]
> Now, it's better. Just a nit pick, keep the gray color for the subheadings which had it earlier. I just wanted you to increase the font size for the blue headings and keep the gray ones as it is.

Changes made

File: src/sections/Gsoc/index.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        </a>
+      </li>
+            <li className="gsoc">
+      <b>Mentee:</b> 
+Kush Trivedi (<a href="https://github.com/kushthedude">@kushthedude</a>)
+</li>
+    </ul>
+  </div>
+  
+
```
Comment (Nikhil-Ladha) [inline src/sections/Gsoc/index.js:117]: Remove these unnecessary blank lines. From anywhere possible in this file.

File: src/sections/Gsoc/gsoc.style.js [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    -moz-user-select: none;
+    -ms-user-select: none;
+    user-select: none;
+    -webkit-tap-highlight-color: transparent;
+    vertical-align: middle;
+    z-index: 1;
+    -webkit-transition: 0.3s ease-out;
+    transition: 0.3s ease-out;
+}
+.btn, .btn-large {
```
Comment (Tanuj22) [inline src/sections/Gsoc/gsoc.style.js:36]: Some some these classes won't be needed now right?

File: src/sections/Gsoc/index.js (hunk 2 above)
Comment (vinayaksh42) [reply src/sections/Gsoc/index.js:117]: okay

File: src/sections/Gsoc/gsoc.style.js (hunk 3 above)
Comment (vinayaksh42) [reply src/sections/Gsoc/gsoc.style.js]: yes right, Removed them.

**Review by Tanuj22 — APPROVED** [review]
LGTM! Good work @vinayaksh42 !

**Review by Nikhil-Ladha — APPROVED** [review]
Good job @vinayaksh42 . Speedy updates, helped us go through it faster. Hope to see more such changes coming from you in future :rocket: 
Sorry, for so nitpicking but that will help you in your next PR :)

**Comment by Nikhil-Ladha** [comment]
Yeahhh!! Your first contribution at Layer5!! :tada:

**Comment by vinayaksh42** [comment]
> Good job @vinayaksh42 . Speedy updates, helped us go through it faster. Hope to see more such changes coming from you in future 🚀
> Sorry, for so nitpicking but that will help you in your next PR :)

Thanks for your patience with me @Tanuj22 @Nikhil-Ladha

**Pull request merged**